src/clarity-angular/data/datagrid/datagrid-column.ts

```diff
@@ -155,13 +155,29 @@ export class DatagridColumn extends DatagridFilterRegistrar<DatagridStringFilter
     /**
      * Comparator to use when sorting the column
      */
-    @Input("clrDgSortBy") public sortBy: Comparator<any>;
+
+    private _sortBy: Comparator<any>;
+
+    public get sortBy() {
+        return this._sortBy;
+    }
+
+    @Input("clrDgSortBy")
+    public set sortBy(comparator: Comparator<any>) {
+        if (comparator) {
+            this._sortBy = comparator;
+        } else {
+            if (!this._field) {
+                delete this._sortBy;
+            }
```

Comment:
I think you're missing a case:
```typescript
column.sortBy = myComparator;
column.field = "hello";
column.sortBy = undefined;
```
This sequence should make the column use the default `DatagridPropertyComparator` from the field, but right now at the end of this sequence, the column is still using `myComparator`, which was **not** deleted. You can write a unit test for it, you'll see. 😉